### PR TITLE
[1LP][RFR] Corrected zone error flash for versions

### PR DESCRIPTION
**Pull request by Yadnyawalkya**
{{ pytest: -v cfme/tests/configure/test_zones.py -k 'test_zone_add_dupe' }}

**What was the issue?**
Wrong flash!

### CODE REVIEW

File: cfme/tests/configure/test_zones.py [hunk 1]
```diff
@@ -77,7 +77,11 @@ def test_zone_add_dupe(request):
         description=description)
     request.addfinalizer(zone.delete)
 
-    with error.expected('Name has already been taken'):
+    if current_appliance.version >= 5.9:
+        error_flash = "Name is not unique within region 0"
```
Comment (psav) [inline cfme/tests/configure/test_zones.py]: **REQUIRED** This should include the region number from the appliance object, and not statically set it

Comment (Yadnyawalkya) [reply cfme/tests/configure/test_zones.py]: @psav, Done! Make sense! 👍

File: cfme/tests/automate/test_service_dialog.py [hunk 2]
```diff
@@ -68,11 +68,12 @@ def test_service_dialog_duplicate_name(appliance, request):
     label = 'duplicate_' + fauxfactory.gen_alphanumeric()
     dialog, element = create_dialog(appliance, element_data, label=label)
     request.addfinalizer(dialog.delete_if_exists)
+    region = appliance.server.zone.region.number
```
Comment (psav) [inline cfme/tests/automate/test_service_dialog.py]: **REQUIRED**: lets call this ```region_number``` or ```region_num```

Comment (Yadnyawalkya) [reply cfme/tests/automate/test_service_dialog.py]: @psav, Done! 👍

**Pull request merged**